GWCS test failure with scipy `1.10.0`

In downstream testing of ASDF I noticed that the newly released `scipy 1.10.0` seems to be causing a test failure.

- Indeed with `scipy 1.10.0` installed I can reproduce the error locally. With the following traceback:

```
__________________________________________________________________________________________________________________ test_to_fits_tab_miri_lrs __________________________________________________________________________________________________________________

    def test_to_fits_tab_miri_lrs():
        af = asdf.open(get_pkg_data_filename('data/miri_lrs_wcs.asdf'))
        w = af.tree['wcs']

        # FITS WCS -TAB:
>       hdr, bt = w.to_fits(sampling=0.25)

gwcs/tests/test_wcs.py:1027:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
gwcs/wcs.py:2447: in to_fits
    hdr, bin_table_hdu = self._to_fits_tab(
gwcs/wcs.py:2631: in _to_fits_tab
    transform(*xyz),
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1136: in __call__
    outputs = self._generic_evaluate(evaluate, inputs, fill_value, with_bbox)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1098: in _generic_evaluate
    outputs = evaluate(_inputs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3352: in evaluate
    return self._evaluate(*_inputs, **kwargs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3380: in _evaluate
    leftval = self.left(*args, **kw)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1136: in __call__
    outputs = self._generic_evaluate(evaluate, inputs, fill_value, with_bbox)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1098: in _generic_evaluate
    outputs = evaluate(_inputs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3352: in evaluate
    return self._evaluate(*_inputs, **kwargs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3380: in _evaluate
    leftval = self.left(*args, **kw)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1136: in __call__
    outputs = self._generic_evaluate(evaluate, inputs, fill_value, with_bbox)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1098: in _generic_evaluate
    outputs = evaluate(_inputs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3352: in evaluate
    return self._evaluate(*_inputs, **kwargs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3395: in _evaluate
    return self.right(*leftval, **kw)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1136: in __call__
    outputs = self._generic_evaluate(evaluate, inputs, fill_value, with_bbox)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1098: in _generic_evaluate
    outputs = evaluate(_inputs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3352: in evaluate
    return self._evaluate(*_inputs, **kwargs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3388: in _evaluate
    rightval = self.right(*(args[self.left.n_inputs :]), **kw)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1136: in __call__
    outputs = self._generic_evaluate(evaluate, inputs, fill_value, with_bbox)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1098: in _generic_evaluate
    outputs = evaluate(_inputs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3352: in evaluate
    return self._evaluate(*_inputs, **kwargs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:3397: in _evaluate
    return self.right(leftval, **kw)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:424: in __call__
    new_call = make_function_with_signature(
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:401: in __call__
    return super(cls, self).__call__(*inputs, **kwargs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1136: in __call__
    outputs = self._generic_evaluate(evaluate, inputs, fill_value, with_bbox)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:1098: in _generic_evaluate
    outputs = evaluate(_inputs)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/core.py:984: in evaluate
    return self.evaluate(*chain(_inputs, parameters))
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/astropy/modeling/tabular.py:242: in evaluate
    result = interpn(
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/scipy/interpolate/_rgi.py:654: in interpn
    return interp(xi)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/scipy/interpolate/_rgi.py:331: in __call__
    indices, norm_distances = self._find_indices(xi.T)
../../../.pyenv/versions/3.10.8/envs/gwcs/lib/python3.10/site-packages/scipy/interpolate/_rgi.py:499: in _find_indices
    return find_indices(self.grid, xi)
_rgi_cython.pyx:95: in scipy.interpolate._rgi_cython.find_indices
    ???
stringsource:660: in View.MemoryView.memoryview_cwrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   ValueError: buffer source array is read-only

stringsource:350: ValueError
```

The `pip freeze` for the failing configuration is:
```
asdf==2.14.3
asdf-astropy==0.3.0
asdf-coordinates-schemas==0.1.0
asdf-standard==1.0.3
asdf-transform-schemas==0.3.0
asdf-unit-schemas==0.1.0
asdf-wcs-schemas==0.1.1
astropy==5.2
attrs==22.2.0
certifi==2022.12.7
charset-normalizer==2.1.1
ci-watson==0.6.1
codecov==2.1.12
coverage==7.0.3
crds==11.16.17
exceptiongroup==1.1.0
filelock==3.9.0
-e git+ssh://git@github.com/WilliamJamieson/gwcs.git@e5c17e7b947043cacae22b79b5f9ec2e8dee0b88#egg=gwcs
hypothesis==6.61.0
idna==3.4
importlib-metadata==6.0.0
iniconfig==1.1.1
jmespath==1.0.1
jsonschema==4.17.3
lxml==4.9.2
numpy==1.24.1
packaging==22.0
Parsley==1.3
pluggy==1.0.0
psutil==5.9.4
pyerfa==2.0.0.1
pyrsistent==0.19.3
pytest==7.2.0
pytest-arraydiff==0.5.0
pytest-astropy==0.10.0
pytest-astropy-header==0.2.2
pytest-cov==4.0.0
pytest-doctestplus==0.12.1
pytest-filter-subpackage==0.1.2
pytest-mock==3.10.0
pytest-openfiles==0.5.0
pytest-remotedata==0.4.0
PyYAML==6.0
requests==2.28.1
scipy==1.10.0
semantic-version==2.10.0
sortedcontainers==2.4.0
tomli==2.0.1
urllib3==1.26.13
zipp==3.11.0
```

- Note that if I install `scipy 1.9.3` (the previous version of `scipy`) the error disappears. The `pip freeze in this case` is

```
asdf==2.14.3
asdf-astropy==0.3.0
asdf-coordinates-schemas==0.1.0
asdf-standard==1.0.3
asdf-transform-schemas==0.3.0
asdf-unit-schemas==0.1.0
asdf-wcs-schemas==0.1.1
astropy==5.2
attrs==22.2.0
certifi==2022.12.7
charset-normalizer==2.1.1
ci-watson==0.6.1
codecov==2.1.12
coverage==7.0.3
crds==11.16.17
exceptiongroup==1.1.0
filelock==3.9.0
-e git+ssh://git@github.com/WilliamJamieson/gwcs.git@e5c17e7b947043cacae22b79b5f9ec2e8dee0b88#egg=gwcs
hypothesis==6.61.0
idna==3.4
importlib-metadata==6.0.0
iniconfig==1.1.1
jmespath==1.0.1
jsonschema==4.17.3
lxml==4.9.2
numpy==1.24.1
packaging==22.0
Parsley==1.3
pluggy==1.0.0
psutil==5.9.4
pyerfa==2.0.0.1
pyrsistent==0.19.3
pytest==7.2.0
pytest-arraydiff==0.5.0
pytest-astropy==0.10.0
pytest-astropy-header==0.2.2
pytest-cov==4.0.0
pytest-doctestplus==0.12.1
pytest-filter-subpackage==0.1.2
pytest-mock==3.10.0
pytest-openfiles==0.5.0
pytest-remotedata==0.4.0
PyYAML==6.0
requests==2.28.1
scipy==1.9.3
semantic-version==2.10.0
sortedcontainers==2.4.0
tomli==2.0.1
urllib3==1.26.13
zipp==3.11.0
```

- I currently do not know if this is an `astropy` bug or if it is directly in `gwcs`.

--- a/setup.cfg
+++ b/setup.cfg
@@ -23,7 +23,7 @@ install_requires =
     asdf >= 2.8.1
     astropy >= 5.1
     numpy
-    scipy
+    scipy < 1.10.0
     asdf_wcs_schemas
     asdf-astropy >= 0.2.0
 

